kernel/src/hil/framebuffer.rs

```diff
@@ -0,0 +1,143 @@
+//! Interface for FrameBuffer
+use crate::returncode::ReturnCode;
+use enum_primitive::cast::FromPrimitive;
+use enum_primitive::enum_from_primitive;
+
+enum_from_primitive! {
+    #[derive(Copy, Clone, PartialEq)]
+    pub enum ScreenRotation {
+        Normal = 0,
+        Rotated90 = 1,
+        Rotated180 = 2,
+        Rotated270 = 3,
+    }
+}
+
+impl From<ScreenRotation> for usize {
+    fn from(rotation: ScreenRotation) -> usize {
+        match rotation {
+            ScreenRotation::Normal => 0,
+            ScreenRotation::Rotated90 => 1,
+            ScreenRotation::Rotated180 => 2,
+            ScreenRotation::Rotated270 => 3,
+        }
+    }
+}
+
+enum_from_primitive! {
+    #[derive(Copy, Clone, PartialEq)]
+    #[allow(non_camel_case_types)]
+    pub enum ScreenColorDepth {
+        None = 0,
+        Mono = 1,
+        RGB_233 = 8,
+        RGB_565 = 16,
+        RGB_888 = 24,
+        ARGB_8888 = 32,
```

Comment:
This also feels like it might be too device-specific for a generic framebuffer HIL (I assume there are other 8-bit encodings that are not `RBG_233`).

Reply:
I changed the enum to ``ScreenPixelFormat`` so that it allows the definition of various pixel formats decoupled from the number of bits. I added some, others can add other formats. I also created an impl for the enum with one function that returns the number of bits per pixel.